commit e5bfb21d98b660580b8cdd8a442d12d38d655d11
Author: Sudeep Holla <sudeep.holla@arm.com>
Date:   Mon May 18 10:12:16 2020 +0100

    firmware: smccc: Add HAVE_ARM_SMCCC_DISCOVERY to identify SMCCC v1.1 and above
    
    SMCCC v1.0 lacked discoverability of version and features. To accelerate
    adoption of few mitigations and protect systems more rapidly from various
    vulnerability, PSCI v1.0 was updated to add SMCCC discovery mechanism
    though the PSCI firmware implementation of PSCI_FEATURES(SMCCC_VERSION)
    which returns success on firmware compliant to SMCCC v1.1 and above.
    
    This inturn makes SMCCC v1.1 and above dependent on ARM_PSCI_FW for
    backward compatibility. Let us introduce a new hidden config for the
    same to build more features on top of SMCCC v1.1 and above.
    
    While at it, also sort alphabetically the psci entry.
    
    Signed-off-by: Sudeep Holla <sudeep.holla@arm.com>
    Tested-by: Etienne Carriere <etienne.carriere@st.com>
    Reviewed-by: Etienne Carriere <etienne.carriere@st.com>
    Acked-by: Mark Rutland <mark.rutland@arm.com>
    Link: https://lore.kernel.org/r/20200518091222.27467-2-sudeep.holla@arm.com
    Signed-off-by: Will Deacon <will@kernel.org>

diff --git a/drivers/firmware/Kconfig b/drivers/firmware/Kconfig
index 8007d4aa76dc..4843e94713a4 100644
--- a/drivers/firmware/Kconfig
+++ b/drivers/firmware/Kconfig
@@ -295,15 +295,13 @@ config TURRIS_MOX_RWTM
 	  other manufacturing data and also utilize the Entropy Bit Generator
 	  for hardware random number generation.
 
-config HAVE_ARM_SMCCC
-	bool
-
-source "drivers/firmware/psci/Kconfig"
 source "drivers/firmware/broadcom/Kconfig"
 source "drivers/firmware/google/Kconfig"
 source "drivers/firmware/efi/Kconfig"
 source "drivers/firmware/imx/Kconfig"
 source "drivers/firmware/meson/Kconfig"
+source "drivers/firmware/psci/Kconfig"
+source "drivers/firmware/smccc/Kconfig"
 source "drivers/firmware/tegra/Kconfig"
 source "drivers/firmware/xilinx/Kconfig"